### PR TITLE
Rep 2379 logging timestamp formatting

### PR DESCRIPTION
**Pull request by Mario-Lopez**
The slf4j http logging string already had support for adding a {} between the %t characters, so I added support for putting a date format in there.  Not specifying a format defaults to the current behavior.  A timebomb was added to change the default format in version 8.  Details about the formatting are documented here (under %{format}t):
https://repose.atlassian.net/wiki/display/REPOSE/SLF4J+HTTP+Logging+filter#SLF4JHTTPLoggingfilter-Whatislogged

Functional tests were added to confirm the existing default behavior and to confirm the custom date format behavior.

### CODE REVIEW

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java [hunk 1]
```diff
@@ -28,8 +29,20 @@
...
+        this.dateFormat = StringUtils.isEmpty(dateFormat) ? DEFAULT_DATE_FORMAT : dateFormat;
+    }
+
     @Override
     public String handle(HttpServletRequest request, HttpServletResponse response) {
-        return new SimpleDateFormat("dd-MM-yyyy-HH:mm:ss.SSS").format(Calendar.getInstance().getTime());
+        return new SimpleDateFormat(dateFormat).format(Calendar.getInstance().getTime());
+    }
+
+    public String getDateFormat() {
```
Comment (adrianjgeorge) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: why is this here?

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java [hunk 2]
```diff
@@ -28,8 +29,20 @@
...
+    private final String dateFormat;
+
+    public TimeReceivedHandler(final String dateFormat) {
+        this.dateFormat = StringUtils.isEmpty(dateFormat) ? DEFAULT_DATE_FORMAT : dateFormat;
+    }
+
     @Override
     public String handle(HttpServletRequest request, HttpServletResponse response) {
-        return new SimpleDateFormat("dd-MM-yyyy-HH:mm:ss.SSS").format(Calendar.getInstance().getTime());
+        return new SimpleDateFormat(dateFormat).format(Calendar.getInstance().getTime());
```
Comment (adrianjgeorge) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: the constructor errors out if it's a bad format right? if so i'd suggest building it in the constructor, that failure should chain up and prevent repose from starting, instead of bombing after the request has been successfully processed by the origin service.

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java (hunk 1 above)
Comment (Mario-Lopez) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: HttpLogFormatterTest.java has some string parsing tests, and unfortunately it's forced to be more like an integration test than a unit test.  The existing tests just throw some data in and then verify it gets the correct data out, but the TimeReceivedHandler code is grabbing the current time instead of using some input data I can manipulate.  To verify the parsing code is working, I can either use this method to compare the the date format strings or I could grab the output of the class and use a regex to see if it looks like the correct date format was used.

Should I remove the method and go with the regex check?

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java (hunk 2 above)
Comment (Mario-Lopez) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: Good call.  I'll add this in.

Comment (Mario-Lopez) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: Update pushed.

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java (hunk 1 above)
Comment (adrianjgeorge) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: yeah, lets go with that. We try to avoid test code bleeding into prod.

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java [hunk 3]
```diff
@@ -28,8 +29,23 @@
...
 
+    private final static String DEFAULT_DATE_FORMAT = "dd-MM-yyyy-HH:mm:ss.SSS";
+
+    private final String dateFormat;
+
+    public TimeReceivedHandler(final String dateFormat) {
+        this.dateFormat = StringUtils.isEmpty(dateFormat) ? DEFAULT_DATE_FORMAT : dateFormat;
+
+        // build it here so an exception will be thrown during application startup if the pattern is invalid
+        new SimpleDateFormat(dateFormat);
```
Comment (adrianjgeorge) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: Not sure why you didn't store it and use it?

Comment (Mario-Lopez) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: SimpleDateFormat is not thread-safe.  I thought about adding locking around it, but I wasn't sure what that would buy us over simply creating a new one each time.

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java (hunk 1 above)
Comment (Mario-Lopez) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: kk, I'll remove the getter.

File: repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java (hunk 3 above)
Comment (adrianjgeorge) [inline repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java]: fair enough
